BraintreeCore/src/test/java/com/braintreepayments/api/core/ConfigurationCacheUnitTest.kt

```diff
@@ -1,62 +1,72 @@
 package com.braintreepayments.api.core
 
-import com.braintreepayments.api.testutils.Fixtures
-import com.braintreepayments.api.core.Configuration.Companion.fromJson
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
 import com.braintreepayments.api.sharedutils.BraintreeSharedPreferences
-import org.robolectric.RobolectricTestRunner
-import io.mockk.*
+import com.braintreepayments.api.testutils.Fixtures
+import org.junit.After
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertNull
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
 import java.util.concurrent.TimeUnit
 
 @RunWith(RobolectricTestRunner::class)
 class ConfigurationCacheUnitTest {
 
-    private var braintreeSharedPreferences: BraintreeSharedPreferences = mockk(relaxed = true)
+    private lateinit var braintreeSharedPreferences: BraintreeSharedPreferences
+
+    @Before
+    fun beforeEach() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        braintreeSharedPreferences = BraintreeSharedPreferences.getInstance(context)
+    }
+
+    @After
+    fun afterEach() {
+        braintreeSharedPreferences.clearSharedPreferences()
+    }
 
     @Test
-    fun saveConfiguration_savesConfigurationInSharedPrefs() {
-        val configuration = fromJson(Fixtures.CONFIGURATION_WITHOUT_ACCESS_TOKEN)
+    fun getConfiguration_returnsReturnsNullByDefault() {
+        val authorization = Authorization.fromString(Fixtures.TOKENIZATION_KEY)
+        val configUrl = "https://sample.com/url"
+
         val sut = ConfigurationCache(braintreeSharedPreferences)
-        sut.saveConfiguration(configuration, "cacheKey", 123L)
-        verify {
-            braintreeSharedPreferences.putStringAndLong(
-                "cacheKey",
-                configuration.toJson(),
-                "cacheKey_timestamp",
-                123L
-            )
-        }
+        assertNull(
+            sut.getConfiguration(authorization, configUrl, TimeUnit.MINUTES.toMillis(20))
+        )
     }
 
     @Test
     fun getConfiguration_returnsConfigurationFromSharedPrefs() {
-        val configuration = fromJson(Fixtures.CONFIGURATION_WITHOUT_ACCESS_TOKEN)
-        every { braintreeSharedPreferences.containsKey("cacheKey_timestamp") } returns true
-        every { braintreeSharedPreferences.getLong("cacheKey_timestamp") } returns 0L
-        every { braintreeSharedPreferences.getString("cacheKey", "") } returns configuration.toJson()
```

Comment:
Also replacing `mockk` `BraintreeSharedPreferences` with a real instance to avoid leaking implementation details of the cache key's string format.